python/requirements/ml/data-requirements.txt

```diff
@@ -14,3 +14,4 @@ modin==0.31.0; python_version >= '3.12'
 pandas==2.2.2; python_version >= '3.12'
 responses==0.13.4
 pymars>=0.8.3; python_version < "3.12"
+pyiceberg==0.7.0
```

Comment:
since `pyiceberg` isn't a critical dependency for running ray data (it is only required for the `read_iceberg()` path), let's remove it from the `data-requirements.txt` file and keep it on `data-test-requirements.txt`.